src/nodes.coffee

```diff
@@ -1081,6 +1150,25 @@ exports.Value = class Value extends Base
 
     fragments
 
+  ast: ->
+    ret = @base.ast()
+    for prop, propIndex in @properties
+      ret =
+        mergeAstLocationData(
+          merge {
+            type: 'MemberExpression'
+            object: ret
+            property: prop.ast()
+            computed: prop instanceof Index or prop.name?.unwrap() not instanceof PropertyName
+            optional: !!prop.soak
+            shorthand: !!prop.shorthand
```

Comment:
`shorthand` is the only "extension" field that you wouldn't find on a Babel-style `MemberExpression` node (with the above caveat about `optional`). It's being used to be able to distinguish shorthand `::` from `.prototype`